server/src/main/java/org/tbbtalent/server/api/admin/CandidateAdminApi.java

```diff
@@ -320,10 +323,13 @@ public void resolveOutstandingTasks(@RequestBody ResolveTaskAssignmentsRequest r
         candidateService.resolveOutstandingTaskAssignments(request);
     }
 
-    @GetMapping("/token/{cn}")
-    public String generateToken(@PathVariable("cn") String candidateNumber) {
-        String token = this.candidateTokenProvider.generateToken(candidateNumber, 365L);
-        return token;
-    }
+    @GetMapping(value = "/token/{cn}", produces = MediaType.TEXT_PLAIN_VALUE)
+    public String generateToken(@PathVariable("cn") String candidateNumber,
+                                @RequestParam(defaultValue = "false") boolean restrictCandidateOccupations,
+                                @RequestParam(defaultValue = "") List<Long> candidateOccupationIds) {
+             CvClaims cvClaims = new CvClaims(candidateNumber, restrictCandidateOccupations, candidateOccupationIds);
+             String token = this.candidateTokenProvider.generateCvToken(cvClaims, 365L);
+             return token;
+        }
```

Comment:
I am thinking that we might need to move away from this token based url. It was my design originally but I think that there is a better way to do it obfuscated urls.

The better way is to create obfuscated urls using some unique string generated from a globally unique id and store the link in a new database table of generated candidate CVs and their generated links. The database would store all the stuff that is currently encoded in the token - ie candidate number, link expiry - plus your new data around restricted candidate occupations. Then when the TC receives processes such a url, it looks up the link in the database to check expiry and regenerate the CV from the data associated with the link.

We end up with shorter - but still obfuscated - urls (which could include candidate number and maybe the job number in plain text). There are other advantages in recording these autogenerated CV links in the TC database.